Add Terraform stack level export task in covalence

disaac opened the pull request:
DEVOPS-2447
* Added stack_export task to terraform stacks
* named the export base directory using the stack.full_name to create separate exports per environment.
* Remove the stack export directory before executing each task. To avoid conflicts and ensure latest data is added to export.
##TODO:
* Create stack export for packer stacks
* Create tests for stack_export functionality.

disaac committed:
Add Terraform stack level export task in covalence

DEVOPS-2447
* Added stack_export task to terraform stacks
* named the export base directory using the stack.full_name to create separate exports per environment.
* Remove the stack export directory before executing each task. To avoid conflicts and ensure latest data is added to export.
##TODO:
* Create stack export for packer stacks
* Crate tests for stack_export functionality.

hinling-whistle commented on lib/covalence/core/services/terraform_stack_tasks.rb:
```diff
@@ -44,6 +44,22 @@ def stack_shell
+        stack_export_init(File.expand_path(File.join(Covalence::STACK_EXPORT,'terraform',stack.full_name))).each do |stackdir|
+          logger.info "Created stackdir: #{stackdir}"
+          populate_workspace(stackdir)
+          Dir.chdir(stackdir) do
```
is it possible not to use `Dir.chdir()`?  Using `Dir.chdir()` makes it really hard to put threading around it.

disaac replied:
@hinling-whistle Updated to no longer use chdir and added path specification to the materialize_state_inputs function defaulting to current working directory.

disaac committed:
Avoiding the use of chdir to remain thread safe

* Updated materialize_state_inputs to accept a path and default to current working directory if one is not specified.
* Updated test to write to tmpdir and then specify path on call to function to verify this.

# Conflicts:
#	Gemfile.lock

hinling-whistle reviewed (approved):
👍

disaac committed:
Adding packer_stack_export

* Added ability to export packer stack tasks as well.
* Removed rake tests on input file validation since this is at a stack level not an environment variable.

disaac committed:
Fix destination typo for stack exports

* Updated the description of tasks with stack exports to have the correct location.
* Increased version number since gem push will fail attempting to overwrite prior release.

# Conflicts:
#	.circleci/config.yml
#	CHANGELOG.md
#	Gemfile.lock
#	lib/covalence/version.rb
